internal/quaiapi/api.go

```diff
@@ -1263,6 +1264,27 @@ func AccessList(ctx context.Context, b Backend, blockNrOrHash rpc.BlockNumberOrH
 		to = *args.To
 	} else {
 		to = crypto.CreateAddress(args.from(nodeLocation), uint64(*args.Nonce), *args.Data, nodeLocation)
+		if _, err := to.InternalAndQuaiAddress(); err != nil {
+			var salt [32]byte
+			binary.BigEndian.PutUint64(salt[24:], uint64(*args.Nonce))
+			// Iterate through possible nonce values to find a suitable contract address.
+			for i := 0; i < params.MaxAddressGrindAttempts; i++ {
+
+				// Place i in the [32]byte array.
+				binary.BigEndian.PutUint64(salt[16:24], uint64(i))
+
+				// Generate a potential contract address.
+				contractAddr := crypto.CreateAddress2(args.from(nodeLocation), salt, crypto.Keccak256Hash(*args.Data).Bytes(), nodeLocation)
```

Comment:
should we be using create address 2 in this case? it should be createadress right for the normal flow of the contract?

Reply:
For grinding, we use CreateAddress2

Reply:
oh okay then

Reply:
but you aren't just using CreateAddress2. You are also grinding the call to CreateAddress2. Why? This grinding logic should not be part of the AccessList() method

Reply:
I'm just replicating the normal Create flow (used to determine the contract address) in evm.go

Reply:
Yeah, thats the point. We shouldn't replicate logic. Put the common logic in a helper function, and call that helper in both places.

Reply:
Fixed @wizeguyy